File: in_Java/1418 Display Table of Food Orders in a Restaurant.java
class Solution {
    public List<List<String>> displayTable(List<List<String>> orders) {
        List<List<String>> res = new ArrayList<>();
        Map<String, List<String>> hm = new HashMap<>();
        Set<String> allfood = new HashSet<>();
        Set<String> tables = new HashSet<>();
        for(List<String> x : orders){
            String tableid = x.get(1);
            tables.add(tableid);
            String food = x.get(2);
            allfood.add(food);
            List<String> cur;
            if(!hm.containsKey(tableid)){
                cur = new ArrayList<>();
                cur.add(food);
                hm.put(tableid, cur);
            }else{
                cur = hm.get(tableid);
                cur.add(food);
                hm.put(tableid,cur);
            }
        }
        String[] foodnames = new String[allfood.size()];
        int id = 0;
        for(String f : allfood) foodnames[id++] = f;
        Arrays.sort(foodnames);
        int[] tablename = new int[tables.size()];
        id = 0;
        for(String t : tables) tablename[id++] = Integer.parseInt(t);
        Arrays.sort(tablename);
        String[] tablenames = new String[tables.size()];
        id = 0;
        for(int p : tablename) tablenames[id] = String.valueOf(tablename[id++]);
                
        List<String> r1 = new ArrayList<>();
        r1.add("Table");
        for(String ff: foodnames) r1.add(ff);
        res.add(r1);
        
        for(int i = 0; i < tablenames.length; i++){
            r1 = new ArrayList<>();
            String cur_table = tablenames[i];
            r1.add(cur_table);
            int[] count = new int[foodnames.length];           
            for(int j = 0; j < foodnames.length; j++){
                String fs = foodnames[j];
                for(String oo : hm.get(cur_table)){
                    if(oo.equals(fs)) count[j]++;
                }   
            }
            for(int vv : count) r1.add(String.valueOf(vv));
            res.add(r1);
        }
        return res;
    }
}

File: in_Python/1418 Display Table of Food Orders in a Restaurant.py
class Solution(object):
    def displayTable(self, orders):
        """
        :type orders: List[List[str]]
        :rtype: List[List[str]]
        """
        dict = {}
        food_list = []
        all_orders = []
        for o in orders:
            order_id = o[1]
            if int(order_id) not in all_orders:
                all_orders.append(int(order_id))
            food = o[2]
            if food not in food_list:
                food_list.append(food)
            if order_id not in dict:
                temp = []
                temp.append(food)
                dict[order_id] = temp
            else:
                temp = dict[order_id]
                temp.append(food)
                dict[order_id] = temp
        food_list.sort()
        all_orders.sort()
        res = []
        for row in all_orders:
            cur = []
            cur.append(str(row))
            foods = dict[str(row)]
            for i in range(len(food_list)):
                cur.append(0)
            for f in foods:
                cur[food_list.index(f)+1] += 1
            for c in range(len(cur)):
                cur[c] = str(cur[c])
            res.append(cur)
        header = food_list
        header.insert(0, 'Table')
        res.insert(0,header)
        return res
        
        